torch/package/_selective_extern.py

```diff
@@ -0,0 +1,14 @@
+SELECTIVE_INTERN_LIST = [
+    $interned_modules
+]
+
+if "__torch_package__" in dir():
+
+    def __getattr__(name):
+        package_name = $package_name
+        if name not in SELECTIVE_INTERN_LIST:
+            return getattr(_extern_copy, name)
+        obj = locals().get(name, None)
```

Comment:
as mentioned elsewhere; the locals/globals here should effectively be empty, we should either go to `_extern_copy` or to the package.